### PR TITLE
[RDY] terminal-title: restore terminal title after quitting

### PR DESCRIPTION
**Pull request by nateozem**
This is a quick implementation to see the change in action.
Do note that this is a test and will see the terminal title with "Thanks
for flying Vim", after quitting vim. This is just a start and I will begin to
fix the issue. Do note with the old implementation, it seems to be
convoluted and can be simplify.
Suggestions are appreciated though.
Relating to #4063 issue and 3baba1e7bc6698e6bc9f1d37fce88b30d6274bc9 commit.

### CODE REVIEW

**Comment by nateozem** [comment]
Maybe I shouldn't use the `--rebase` option when I execute the pull operation to my working branch, so I can merge changes from master.
I could recreate a new pull request with only my changes.

**Comment by teto** [comment]
git's hash depends on the previous commit. You can $ git rebase upstream/master 
or from you dev branch $ git rebase -i @~<YourFirstRev> master.
Then git push -f on this branch. No need to open a new PR

**Comment by nateozem** [comment]
@teto thanks. I'll do that next time.

**Comment by justinmk** [comment]
are you still working on this?

**Comment by nateozem** [comment]
I thought someone would comment on it. I can change it to ready.

File: src/nvim/os_unix.c [hunk 1]
```diff
@@ -138,9 +138,9 @@ void mch_exit(int r) FUNC_ATTR_NORETURN
 {
   exiting = true;
 
-  if (p_title
-      && p_titlestring != NULL
-      && STRLEN(p_titlestring) == 0
+  if ((p_title
+       || (did_enable_title
+           && (p_titlestring == NULL || STRLEN(p_titlestring) == 0)))
       && p_titleold != NULL) {
```
Comment (justinmk) [inline src/nvim/os_unix.c]: @nateozem why is `did_enable_title` needed? If 'title' is enabled and 'titleold' is non-empty, that should be enough. And we already check that anyways ...

Comment (nateozem) [reply src/nvim/os_unix.c]: In the docs, it mention that if `titlestring` is not empty then `titleold` would be used.
```
       'titleold'		string	(default "Thanks for flying Vim")
 			global
 			{only available when compiled with the |+title|
 			feature}
 	This option will be used for the window title when exiting Vim if the
 	original title cannot be restored.  Only happens if 'title' is on or
 	'titlestring' is not empty.
```

In Vim 8.0, if `title` was ever on (even if turned back to off), and the `titlestring` is not empty, then the terminal title is changed to `titleold`. We can make things simple as you have pointed out, but we would need to change the description for `titleold`. The wording in the docs might be a mistake. I'm just not sure the thought process of the author.

Comment (nateozem) [reply src/nvim/os_unix.c]: Please let me know if I'm reading it wrong. I do agree with you to check for only `title` and `titleold`.

Comment (justinmk) [reply src/nvim/os_unix.c]: @nateozem I've made that change, try https://github.com/neovim/neovim/pull/7358

**Comment by justinmk** [comment]
merged in https://github.com/neovim/neovim/pull/7358

File: src/nvim/os_unix.c (hunk 1 above)
Comment (nateozem) [reply src/nvim/os_unix.c]: Nice, I'll give it a try.

**Pull request closed without merging**